tests/device_tests/test_firmware_dump.py

```diff
@@ -0,0 +1,27 @@
+from hashlib import blake2s
+
+from trezorlib import firmware
+from trezorlib.debuglink import TrezorClientDebugLink as Client
+
+FIRMWARE_LENGTHS = {
+    "1": 7 * 128 * 1024 + 64 * 1024,
+    "T": 13 * 128 * 1024,
+}
+
+
+def test_firmware_dump_hash(client: Client) -> None:
+    data = firmware.get_firmware(client)
+    assert len(data) == FIRMWARE_LENGTHS[client.features.model]
+
+    if client.features.fw_vendor != "EMULATOR":
+        # check that the dumped data is not empty
+        assert not all(byte == "\xff" for byte in data)
```

Comment:
We should try to check this against the actual firmware `.bin` file. Can be done later after this PR is merged.

Reply:
Exactly, also we should fix the core emulator to not send all empty data. T1 emulator doesn't seem to do that.

Reply:
...except it does, check fixed in [badb0e7](https://github.com/trezor/trezor-firmware/pull/2248/commits/badb0e71d64d23329b646862db8f872ef3aade92) and reporting added in [ea3db2a](https://github.com/trezor/trezor-firmware/pull/2248/commits/ea3db2adbfe1a9d7ab15bb6a73e6a08ca3b54b77)